src/test/stepFunctions/commands/visualizeStateMachine.test.ts

```diff
@@ -0,0 +1,464 @@
+        ext.visualizationResourcePaths.localWebviewScriptsPath = oldWebviewScriptsPath
+        ext.visualizationResourcePaths.webviewBodyScript = oldWebviewBodyPath
+        ext.visualizationResourcePaths.visualizationLibraryCachePath = oldCachePath
+        ext.visualizationResourcePaths.visualizationLibraryScript = oldScriptPath
+        ext.visualizationResourcePaths.visualizationLibraryCSS = oldCssPath
+        ext.visualizationResourcePaths.stateMachineCustomThemePath = oldThemePath
+        ext.visualizationResourcePaths.stateMachineCustomThemeCSS = oldThemeCssPath
+    })
+
+    // Tests
+    it('Test AslVisualization on setup all properties are correct', async () => {
+        const vis = new MockAslVisualization(mockTextDocumentOne)
+
+        assert.deepStrictEqual(vis.documentUri, mockTextDocumentOne.uri)
+        assert.strictEqual(vis.getIsPanelDisposed(), false)
+        assert.strictEqual(vis.getDisposables().length, 4)
+
+        let panel = vis.getPanel()
+        assert.ok(panel)
+        panel = <vscode.WebviewPanel>panel
+        assert.ok(panel.title.length > 0)
+        assert.strictEqual(panel.viewType, 'stateMachineVisualization')
+
+        let webview = vis.getWebview()
+        assert.ok(webview)
+        webview = <vscode.Webview>webview
+        assert.ok(webview.html)
+    })
+
+    it('Test AslVisualizationManager on setup managedVisualizations set is empty', async () => {
+        assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 0)
+    })
+
+    it('Test AslVisualizationManager managedVisualizations set still empty if no active text editor', async () => {
+        assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 0)
+
+        try {
+            // Preview with no active text editor
+            await aslVisualizationManager.visualizeStateMachine(mockGlobalStorage)
+            assert.fail('Error should be thrown if no active text editor')
+        } catch (error) {
+            assert.ok(error instanceof Error)
+            const errorCasted = <Error>error
+            assert.strictEqual(errorCasted.message, 'Could not get active text editor for state machine render.')
+            assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 0)
+        }
+    })
+
+    it('Test AslVisualizationManager managedVisualizations set has one AslVis on first preview', async () => {
+        assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 0)
+
+        try {
+            // Preview Doc1
+            mockVsCode.showTextDocument(mockTextDocumentOne)
+            await aslVisualizationManager.visualizeStateMachine(mockGlobalStorage)
+            assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 1)
+
+            const managedVisualizations = aslVisualizationManager.getManagedVisualizations()
+            assert.ok(managedVisualizations.get(mockTextDocumentOne.uri.path))
+        } catch (error) {
```

Comment:
Are you expecting an error in this test? If the try block doesn't throw an error, the catch block is never executed (and your test then passes), so something isn't as intended in this test.